controllers/capabilities/tenant_controller.go

```diff
@@ -73,17 +83,6 @@ func (r *TenantReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
-	changed := tenantR.SetDefaults()
-	if changed {
-		err = r.Client.Update(context.TODO(), tenantR)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		reqLogger.Info("Tenant resource updated with defaults")
-		// Expect for re-trigger
-		return ctrl.Result{}, nil
-	}
-
 	masterAccessToken, err := r.FetchMasterCredentials(r.Client, tenantR)
```

Comment:
Reason why it's done that way is because the deletion required the portaClient and finalizers can only be deleted if the portaClient tenant delete succeeds. If we are to do it the other way around, we are risking deleting finalizers before tenant is removed from 3scale. How do you think we could resolve this?

Reply:
Right, I changed my mind. It is correct as it is now. 

If there is an error building the portaClient (involves reading a secret), it will retry. 

The only "edge" case here is that is some other controller adds its own finalizer, our controller will still read secrets and build portaClient even when the delete job is done and our finalizer is removed. But a small issue IMO.